R/addins.R

```diff
@@ -97,77 +97,37 @@ get_rstudio_context <- function() {
   rstudioapi::getActiveDocumentContext()
 }
 
-
-# Dedicated binding for package styling addin. Simple wrapper calling style_pkg
-# with the selected addins style.
-style_package <- function() {
-  communicate_addins_style()
-  style_pkg(style = get_addins_style_fun())
-}
-
-
-# `match.fun`-like utility covering "ns::name".
-exported_value_rx <- "^([^:]+)::([^:]+)$"
-is_exported_value <- function(x) {
-  rlang::is_scalar_character(x) && grepl(exported_value_rx, x)
-}
-extract_exported_ns <- function(x) {
-  sub(exported_value_rx, "\\1", x)
-}
-extract_exported_name <- function(x) {
-  sub(exported_value_rx, "\\2", x)
-}
-match_fun <- function(x) {
-  if (is_exported_value(x)) {
-    x <-
-      getExportedValue(
-        extract_exported_ns(x),
-        extract_exported_name(x)
-      )
-  }
-  match.fun(x)
-}
-
-# Binding for style-setting addin.
+#' Ask the user to supply a style
+#'
+#' @keywords internal
 prompt_style <- function() {
-  current_style <- get_addins_style()
+  current_style <- get_addins_style_name()
   new_style <-
     rstudioapi::showPrompt(
       "Select a style",
       "Enter the name of a style function, e.g. `styler::tidyverse_style`",
       current_style
     )
   if (!is.null(new_style)) {
-    set_addins_style(new_style)
+    options(styler.addins.style = new_style)
   }
   invisible(current_style)
 }
 
-# Set/get style used by the addins.
-set_addins_style <- function(style_name) {
-  # match_fun ensures the provided name is a valid function
-  invisible(match_fun(style_name))
```

Comment:
The original idea behind calling `invisible(match_fun(...))` in `set_addings_style()` was to leverage on the side-effect of triggering an error if the provided style does not match a function, *without* setting the new style.

If we want to keep the same logic and behavior, you may consider adding something similar before setting `options(styler.addins.style = new_style)`
```r
    # basic check prior to setting the new_style, making sure we can
    # successfully parse it to a function object
    options(styler.addins.style = new_style)
    # alternative via an explicit assertion and nicer error
    assertthat::assert_that(is.function(eval(parse(text = new_style))), 
      msg = paste(dQuote(new_style), "is not a function")
    )
```
You can check the behavior with `new_style` being e.g. `"non_existing_style"`, `"styler::dummy_style"`, `"mtcars"`, `"non parseable"`.

Reply:
Ok, that is actually a good point. Thanks. I think instead of adding the assertthat dependency, we can write a custom `tryCatch()`. See 320ca363 for how I solved it. What do you think?

Reply:
Sure, if you do not have assertthat dependency yet let's avoid it, see below for my feedback.